### PR TITLE
[8.1] feat (Framework): remove setup

### PR DESCRIPTION
**Pull request by fstagni**
BEGINRELEASENOTES

*Framework
CHANGE: Removed Setup from InstalledComponentsDB
CHANGE: Removing the possibility of multiple setups in one server

ENDRELEASENOTES

### CODE REVIEW

File: src/DIRAC/FrameworkSystem/Agent/ComponentSupervisionAgent.py [hunk 1]
```diff
@@ -121,7 +121,7 @@ def logError(self, errStr, varMsg=""):
 
     def beginExecution(self):
         """Reload the configurations before every cycle."""
-        self.setup = self.am_getOption("Setup", self.setup)
+        self.setup = CSGlobals.getSetup()
```
Comment (andresailer) [inline src/DIRAC/FrameworkSystem/Agent/ComponentSupervisionAgent.py]: the `self.setup` can just be removed now, I think.

File: src/DIRAC/FrameworkSystem/API/AuthHandler.py [hunk 2]
```diff
@@ -80,6 +80,7 @@ def get_index(self):
           }
         """
         if self.request.method == "GET":
+            # FIXME: what is this "setups" for?
```
Comment (aldbr) [inline src/DIRAC/FrameworkSystem/API/AuthHandler.py]: Seems like it is just extra server metadata specific to DIRAC published in the `.well-known` endpoint.
It does not appear in the list of expected authorization server metadata: https://www.rfc-editor.org/rfc/rfc8414#section-2

But it is used by DIRAC here: 
https://github.com/DIRACGrid/DIRAC/blob/85a8918df0cdd1c0a03123f7d746dda060becffa/src/DIRAC/Core/scripts/dirac_configure.py#L332-L333

From what I understand, values are retrieved in the `well-known` endpoint by the `AuthorizationServerMetadata` instance here: https://github.com/DIRACGrid/DIRAC/blob/85a8918df0cdd1c0a03123f7d746dda060becffa/src/DIRAC/FrameworkSystem/private/authorization/utils/Utilities.py#L40

In `dirac-configure`, the `setup` parameter is used to log the current setup: https://github.com/DIRACGrid/DIRAC/blob/85a8918df0cdd1c0a03123f7d746dda060becffa/src/DIRAC/Core/scripts/dirac_configure.py#L476-L477

File: src/DIRAC/FrameworkSystem/scripts/dirac_admin_update_pilot.py [hunk 3]
```diff
@@ -39,26 +38,21 @@ def updatePilot(version, vo):
...
         """
-        setup = vo
-        if not vo:
-            setup = gConfig.getValue("/DIRAC/DefaultSetup")
-        if not setup:
-            return S_ERROR("No value set for /DIRAC/DefaultSetup in CS")
-
-        pilotVersion = gConfig.getValue("Operations/%s/Pilot/Version" % setup, [])
+        # FIXME: use Operations() object
+        pilotVersion = gConfig.getValue("Operations/%s/Pilot/Version" % vo, [])
```
Comment (aldbr) [inline src/DIRAC/FrameworkSystem/scripts/dirac_admin_update_pilot.py]:
```suggestion
        pilotVersion = gConfig.getValue(f"Operations/{vo}/Pilot/Version", [])
```

File: src/DIRAC/FrameworkSystem/Agent/ComponentSupervisionAgent.py [hunk 4]
```diff
@@ -565,7 +559,7 @@ def _checkServiceURL(self, serviceName, options):
         system = options["System"]
         module = options["Module"]
         self.log.info(f"Checking URLs for {system}/{module}")
-        urlsConfigPath = Path.cfgPath(PathFinder.getSystemURLSection(system=system, setup=self.setup), module)
+        urlsConfigPath = Path.cfgPath(PathFinder.getSystemURLSection(system=system), module)
         urls = gConfig.getValue(urlsConfigPath, [])
         self.log.debug(f"Found configured URLs for {module}: {urls}")
         self.log.debug("This URL is %s" % url)
```
Comment (aldbr) [inline src/DIRAC/FrameworkSystem/Agent/ComponentSupervisionAgent.py]:
```suggestion
        self.log.debug("This URL is", url)
```

File: src/DIRAC/FrameworkSystem/scripts/dirac_admin_update_pilot.py [hunk 5]
```diff
@@ -39,26 +38,21 @@ def updatePilot(version, vo):
...
         if not pilotVersion:
-            return S_ERROR("No pilot version set under Operations/%s/Pilot/Version in CS" % setup)
+            return S_ERROR("No pilot version set under Operations/%s/Pilot/Version in CS" % vo)
```
Comment (aldbr) [inline src/DIRAC/FrameworkSystem/scripts/dirac_admin_update_pilot.py]:
```suggestion
            return S_ERROR(f"No pilot version set under Operations/{vo}/Pilot/Version in CS")
```

File: src/DIRAC/FrameworkSystem/scripts/dirac_admin_update_pilot.py [hunk 6]
```diff
@@ -39,26 +38,21 @@ def updatePilot(version, vo):
...
 
         pilotVersion.pop()
         pilotVersion.insert(0, version)
         api = CSAPI()
-        api.setOption("Operations/%s/Pilot/Version" % setup, ", ".join(pilotVersion))
+        api.setOption("Operations/%s/Pilot/Version" % vo, ", ".join(pilotVersion))
```
Comment (aldbr) [inline src/DIRAC/FrameworkSystem/scripts/dirac_admin_update_pilot.py]:
```suggestion
        api.setOption(f"Operations/{vo}/Pilot/Version", ", ".join(pilotVersion))
```

File: src/DIRAC/FrameworkSystem/scripts/dirac_admin_update_pilot.py [hunk 7]
```diff
@@ -39,26 +38,21 @@ def updatePilot(version, vo):
...
         result = api.commit()
         if not result["OK"]:
             gLogger.fatal("Could not commit new version of pilot!")
             return result
 
-        newVersion = gConfig.getValue("Operations/%s/Pilot/Version" % setup)
+        newVersion = gConfig.getValue("Operations/%s/Pilot/Version" % vo)
```
Comment (aldbr) [inline src/DIRAC/FrameworkSystem/scripts/dirac_admin_update_pilot.py]:
```suggestion
        newVersion = gConfig.getValue(f"Operations/{vo}/Pilot/Version")
```

**Comment by fstagni** [comment]
Review OK now?

**Comment by aldbr** [comment]
> Review OK now?

You have not modified `dirac-configure` to remove `setup`, I assume that you want to do it later in another pull request, don't you?
Other than that, it looks OK to me.

**Comment by fstagni** [comment]
> > Review OK now?
> 
> You have not modified `dirac-configure` to remove `setup`, I assume that you want to do it later in another pull request, don't you? Other than that, it looks OK to me.

Correct.

**Comment by aldbr** [comment]
Seems like the test failure is related to this: https://github.com/DIRACGrid/DIRAC/blob/integration/src/DIRAC/ConfigurationSystem/Client/PathFinder.py#L43

`getSystemInstance()` should probably try both:
- `optionPath = Path.cfgPath("/DIRAC/Setups", setup or getDIRACSetup(), system)`
- `optionPath = Path.cfgPath("/DIRAC/", system)`

What do you think?

**Comment by fstagni** [comment]
> Seems like the test failure is related to this: https://github.com/DIRACGrid/DIRAC/blob/integration/src/DIRAC/ConfigurationSystem/Client/PathFinder.py#L43

That line was changed 1 year ago... And this test was not failing 1 week ago so must be related to some other recent commit.

File: src/DIRAC/FrameworkSystem/API/AuthHandler.py (hunk 2 above)
Comment (fstagni) [reply src/DIRAC/FrameworkSystem/API/AuthHandler.py]: This will be deferred to another PR.

**Comment by fstagni** [comment]
Last review? This PR will be merged by the end of the day.

**Comment by fstagni** [comment]
> Last review? This PR will be merged by the end of the day.

Merging now.

**Pull request merged**